src/mongo/db/auth/external/sasl_oidc_server_mechanism.h

```diff
@@ -48,6 +49,11 @@ Copyright (C) 2025-present Percona and/or its affiliates. All rights reserved.
 #include "mongo/db/operation_context.h"
 
 namespace mongo {
+
+namespace crypto {
+class JWSValidatedToken;
+}  // namespace crypto
+
 class SaslOidcServerMechanism final : public MakeServerMechanism<OidcPolicy> {
```

Comment:
I think we should also override the `ServerMechanismBase::getExpirationTime` to return an appropriate expiration time from the token, otherwise the user will be authenticated indefinitely on the session. I can implement that in a separate PR.